daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -82,6 +83,18 @@ abstract class SequenceParserBase(
 
     val infosetIndexStart = pstate.infoset.asInstanceOf[DIComplex].childNodes.size
 
+    if (!isOrdered) {
+      // If this is an unordered sequence, upon completion of parsing all the
+      // elements we will reorder the elements into schema definition order.
+      // This means that we cannot let the infoset walker walk any of the
+      // elements while we are parsing because their event order might change.
+      // To ensure we don't walk, add a block on the parent of the infoset
+      // elements. The infoset walker will inspect this to see if it should
+      // walk any children. We'll remove this block once the unordered sequence
+      // is complete.
+      pstate.infoset.infosetWalkerBlockCount += 1
```

Comment:
So this means we can't incrementally parse an unordered sequence at all, even if it arrives in perfect order. I'm fine with that. We need to accumulate these limitations in a user's guide/note so as to explain why schemas can stream or not. 

We might even want a tunable that turns on "streamability" warnings, and issue SDW for non-streamable constructs in schemas.

Reply:
Agreed. I was also thinking another such warning could be if a point of uncertainty isn't resolved by the time the parsers unwind back to it. This implies there was no discriminator which makes streaming harder. I'll add a bug to add some warnings.